tools/linter/adapters/_linter.py

```diff
@@ -12,29 +12,31 @@
 from functools import cached_property
 from pathlib import Path
 from tokenize import generate_tokens, TokenInfo
-from typing import Any, TYPE_CHECKING
-from typing_extensions import Never
+from typing import Any, Generic, get_args, TYPE_CHECKING
+from typing_extensions import Never, Self, TypeVar
 
 
 if TYPE_CHECKING:
     from collections.abc import Iterator, Sequence
 
 
-FSTRING_START = getattr(token, "FSTRING_START", None)  # py3.12+
-FSTRING_END = getattr(token, "FSTRING_END", None)
-EMPTY_TOKENS = dict.fromkeys(
-    [
-        token.COMMENT,
-        token.DEDENT,
-        token.ENCODING,
-        token.INDENT,
-        token.NEWLINE,
-        token.NL,
-    ]
-)
+# Python 3.12 and up have two new token types, FSTRING_START and FSTRING_END
+NO_TOKEN = -1
+FSTRING_START: int = getattr(token, "FSTRING_START", NO_TOKEN)
+FSTRING_END: int = getattr(token, "FSTRING_END", NO_TOKEN)
+
+START_OF_LINE_TOKENS = {token.DEDENT, token.INDENT, token.NEWLINE}
+IGNORED_TOKENS = {token.COMMENT, token.ENDMARKER, token.ENCODING, token.NL}
+EMPTY_TOKENS = START_OF_LINE_TOKENS | IGNORED_TOKENS
+
 BRACKETS = {"{": "}", "(": ")", "[": "]"}
 BRACKETS_INV = {j: i for i, j in BRACKETS.items()}
 
+ROOT = Path(__file__).absolute().parents[3]
+assert ROOT.name == "pytorch", (
```

Comment:
People don't need to name their pytorch checkout pytorch, you can use whatever folder name you like.  (In fbcode it is not pytorch.)

Reply:
Oops, I knew that. That was left over from development, good catch, @jansel